### ISSUE TITLE
Unable to update contentlets from content import when you have relationship fields

### ISSUE THREAD

**Issue by bryanboza** [dotCMS/core#16095]
In the previous version, when you select some key to import and this already exists, we update the existing content with the new data. 

## Expected Behavior
We need be able to update contents if the key already exists without problems, in case of relationships, we need to delete the existing ones and create new ones. 

## Current Behavior
The system isn't deleting the existing ones and always is trying to create a new one, then depending on the cardinality of the relationship we have an error on logs and we are unable to update the content 
Error: 
```
16:05:10.152  ERROR business.ESContentletAPIImpl - ERROR! Parent content [] cannot be related to child content [1c6c0444-a10d-4871-9ac6-a51e218b785f] because it is already related to parent content [68af894e-3272-48eb-869e-5ff4789b0cfb]
16:05:10.152  ERROR util.ImportUtil - Line #2 contains errors

com.dotmarketing.exception.DotRuntimeException: Line #2 contains errors

	at com.dotmarketing.util.ImportUtil.importLine(ImportUtil.java:1313) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.dotmarketing.util.ImportUtil.importFile(ImportUtil.java:255) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.dotmarketing.portlets.contentlet.action.ImportContentletsAction._generatePreview(ImportContentletsAction.java:466) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.dotmarketing.portlets.contentlet.action.ImportContentletsAction.processAction(ImportContentletsAction.java:182) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portal.struts.PortletRequestProcessor.process(PortletRequestProcessor.java:125) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portlet.StrutsPortlet.processAction(StrutsPortlet.java:113) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portlet.CachePortlet.processAction(CachePortlet.java:130) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portal.action.LayoutAction._processPortletRequest(LayoutAction.java:230) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portal.action.LayoutAction._processActionRequest(LayoutAction.java:247) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portal.action.LayoutAction.execute(LayoutAction.java:89) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:421) ~[struts-1.2.4.jar:1.2.4]
	at org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:226) ~[struts-1.2.4.jar:1.2.4]
	at com.liferay.portal.struts.PortalRequestProcessor.process(PortalRequestProcessor.java:151) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.struts.action.ActionServlet.process(ActionServlet.java:1164) ~[struts-1.2.4.jar:1.2.4]
	at org.apache.struts.action.ActionServlet.doPost(ActionServlet.java:415) ~[struts-1.2.4.jar:1.2.4]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:661) ~[servlet-api.jar:?]
	at com.liferay.portal.servlet.MainServlet.callParentService(MainServlet.java:272) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.liferay.portal.servlet.MainServlet.service(MainServlet.java:481) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.repackage.com.liferay.filters.secure.SecureFilter.doFilter(SecureFilter.java:135) ~[dot.secure-filter-ukv_2.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:199) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:56) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.cms.urlmap.filters.URLMapFilter.doFilter(URLMapFilter.java:114) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.VanityURLFilter.doFilter(VanityURLFilter.java:104) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:399) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:134) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:46) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:102) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.0.0_f0d86aa.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) ~[catalina.jar:8.5.32]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) ~[catalina.jar:8.5.32]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) ~[catalina.jar:8.5.32]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) ~[catalina.jar:8.5.32]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:8.5.32]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1471) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:8.5.32]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_191]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_191]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:8.5.32]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_191]
```
**FYI: Tested on master with the legacy relationships and works as expected** 

## Steps to Reproduce (for bugs)
1. Create two new content types (Movie, Rating)
Movie (text: ID, text:Name, relationship:MovieRating)
Rating (text: ID, text:rating)
3. Create a new piece of content on the movie content type and relate with a piece in rating
4. Now export the contents of movies
5. Edit the name in the CSV file and try to import again
6. Make sure you are able to update without problems 

## Your Environment
Tested on master // Postgres // FF

**Comment by nollymar** [comment]
I was not able to reproduce this issue with the described steps

**Comment by bryanboza** [comment]
I'm able to reproduce exactly with those steps
Here a video: https://screencast.com/t/RLkSRkC8F

**Comment by nollymar** [comment]
This is not an issue. The error is thrown because the contentlet is not being updated but inserted. The key id specified in the csv file does not correspond to the existing contentlet, so the app tries to create a new piece of content and relate it to another contentlet that is already in a relationship (one to many cardinality violation). This case was clarified with @bryanboza in a call.

**Comment by erickgonzalez** [comment]
Needs Work:

- [ ] When updating a content, if it has a **One to One** relationship, throws a Cardinality error but that is not happening.

- [ ] When updating a content, if a **Many to Many** relationship has no values, and you try to relate content via CSV throws an error 

```
WARN  action.ImportContentletsAction - java.lang.NullPointerException
java.lang.NullPointerException: null
	at com.dotmarketing.portlets.contentlet.action.ImportContentletsAction$1.run_aroundBody0(ImportContentletsAction.java:222) ~[dotcms_5.0.0_898f98f.jar:?]
```

**Comment by nollymar** [comment]
Fixed here: https://github.com/dotCMS/core/pull/16191

**Comment by erickgonzalez** [comment]
Tested and working as it should now, no error is thrown.

**Comment by bryanboza** [comment]
Fixed, tested on release-5.0.1 // H2 // FF
